Hw 4

NPavlovaaa opened the pull request.

NPavlovaaa committed:
Stores
Create ProductStore using mobx, add models and collection for it

NPavlovaaa committed:
Stores
Add methods to get data for the ProductPage from the store

NPavlovaaa committed:
Stores
Create FilterStore and add methods for it; fix bugs

NPavlovaaa committed:
Stores
Create SearchStore and add methods for it; transfer product filtering to the store; fix bugs

NPavlovaaa committed:
Stores
Create QueryParamsStore and add methods for it; remove FilterStore and SearchStore; fix bugs

NPavlovaaa committed:
Stores
Transfer the logic of filtering and searching for products to the request for fetching a list of products

ADdinathe commented on src/App/pages/MainPage/components/FIlters/Filters.module.scss:
```diff
@@ -2,8 +2,8 @@
     @include custom_display(column, none);
-    width: 300px;
-    flex: 1;
+    //width: 300px;
+    //flex: 1;
```
лишние коменты убираем

ADdinathe commented on package.json:
```diff
@@ -11,9 +11,13 @@
     "lint:fix": "eslint \"./src/**/*.{js,jsx,ts,tsx}\" --fix"
   },
   "dependencies": {
+    "@types/qs": "^6.9.8",
```
Та же ошибка что и у остальных

ты делаешь один запрос на старте, получаешь весь список товаров и дальше на фронте его фильтруешь и пагинируешь. Это НЕ круто! Нужно использовать параметры запроса к апишке, ниже пример из документации.
https://fakeapi.platzi.com/en/rest/products#:~:text=true-,Pagination,-APIs%20that%20use

Пример работы логики. Пользователь заходит на сайт, ты запрашиваешь кусочек информации (0-10) товаров.
Если пользователь вбивает поисковый запрос, то передаешь поисковую строку. Если пользователь листает список пагинации дальше, запрашиваешь список дальше.

когда делаю первый поисковый запрос найденные карточки товаров не отображаются 
![image](https://github.com/NPavlovaaa/ecommerce/assets/55657904/46ca5a33-8e98-4935-8778-307261a83319)

по поводу проблемы с фильтрами в квери строке, загляни в консоль там ошибки сыпятся при добавлении. Возможно это поможет.

NPavlovaaa committed:
Additional functionality
Create CartPage component, Slider and Spinner components; add property "textView" to Button components; fix bugs

NPavlovaaa committed:
Fix bugs
Change MultiDropDown component;rewrite pagination and getting product list data using offset and limit params; add onClick to Icon component

NPavlovaaa committed:
Fix bugs
